diff --git a/init/Kconfig b/init/Kconfig
index de92d96..4be4a55 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1385,7 +1385,8 @@ config BOOT_CONFIG
 
 choice
 	prompt "Compiler optimization level"
-	default CC_OPTIMIZE_FOR_PERFORMANCE
+	default CC_OPTIMIZE_FOR_PERFORMANCE_O3 if (GCC_VERSION >= 100000 || CC_IS_CLANG)
+	default CC_OPTIMIZE_FOR_PERFORMANCE if GCC_VERSION < 100000
 
 config CC_OPTIMIZE_FOR_PERFORMANCE
 	bool "Optimize for performance (-O2)"
@@ -1396,7 +1397,6 @@ config CC_OPTIMIZE_FOR_PERFORMANCE
 
 config CC_OPTIMIZE_FOR_PERFORMANCE_O3
 	bool "Optimize more for performance (-O3)"
-	depends on ARC
 	help
 	  Choosing this option will pass "-O3" to your compiler to optimize
 	  the kernel yet more for performance.
