diff --git a/kernel/sched/alt_core.c b/kernel/sched/alt_core.c
index 7c998e6..95a2f64 100644
--- a/kernel/sched/alt_core.c
+++ b/kernel/sched/alt_core.c
@@ -3468,6 +3468,7 @@ static inline void update_curr(struct rq *rq, struct task_struct *p)
 	s64 ns = rq->clock_task - p->last_ran;
 
 	p->sched_time += ns;
+    cgroup_account_cputime(p, ns);
 	account_group_exec_runtime(p, ns);
 
 	p->time_slice -= ns;
